### ISSUE TITLE
Regression in typealias with Union{}

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/julia#16922]
This used to work on 0.4.5:

``` julia
julia> typealias VecOrNullableVec{T} Union{AbstractVector{T}, AbstractVector{Nullable{T}}}
AbstractArray{T,1}

julia> VecOrNullableVec{Int}
AbstractArray{Int64,1}

julia> AbstractArray{Nullable{Int},1} <: VecOrNullableVec{Int}
false
```

**Comment by JeffBezanson** [comment]
Caused by 0ffba4018f6e12b97c220a59e6d16776bb8a6314

**Comment by JeffBezanson** [comment]
closed by #18457
